mita/openstack.py

```diff
@@ -111,80 +100,84 @@ def get_driver(self, **kw):
+        return images[0]
+
+    def _get_flavor_by_name(self) -> NodeSize:
+        """Return the flavor reference."""
+        logger.info("Gathering %s flavor details.", self.vm_size)
+
+        for flavor in self.driver.list_sizes():
+            if flavor.name == self.vm_size:
+                return flavor
+
+        raise ResourceNotFound(f"No matching {self.vm_size} flavor found.")
+
+    def _get_network_by_name(self, name: str) -> OpenStackNetwork:
         """
-        Get Apache Libcloud OpenStack driver for nova version 2.0.
+        Return the OpenStackNetwork object.
 
-        https://libcloud.readthedocs.io/en/latest/compute/drivers/
-        openstack.html#compute-2-0-api-version-current
+        Args:
+            name:   The name of the network to be retrieved.
 
         Returns:
-            OpenStack driver
+            An instance of OpenStackNetwork object.
+
+        Raises:
+            ResourceNotFound    when the object cannot be found.
         """
-        return self.driver_v2 if self.driver_v2 else self.get_driver(api_version="2.2")
+        logger.info("Retrieving %s network details.", name)
 
-    def _get_image(self) -> NodeImage:
-        """Return the glance image reference."""
-        try:
-            return [
-                i for i in self.driver_v2.list_images() if i.name == self.image_name
-            ][0]
-        except IndexError:
-            raise ResourceNotFound("Image {} not found".format(self.image_name))
-        except BaseException as be:  # noqa
-            logger.error(be)
-            raise OpenStackDriverError("Encountered an unknown exception.")
+        url = f"{self.driver._networks_url_prefix}?name={name}"
+        _object = self.driver.network_connection.request(url).object
+        networks = self.driver._to_networks(_object)
 
-    def _get_flavor(self) -> NodeSize:
-        """Return the flavor reference."""
-        try:
-            return [f for f in self.driver_v2.list_sizes() if f.name == self.vm_size][0]
-        except IndexError:
-            raise ResourceNotFound("Flavor {} not found".format(self.vm_size))
-        except BaseException as be:  # noqa
-            logger.error(be)
-            raise OpenStackDriverError("Encountered an unknown exception.")
+        if len(networks) != 1:
+            raise ResourceNotFound(f"Exact match failed for {name} network.")
+
+        return networks[0]
 
-    def _has_free_ip_address(self, network: str) -> bool:
+    def _has_free_ip_address(self, net: OpenStackNetwork) -> bool:
         """
         Return True if the given network has an IPv4 Address to lease.
 
         Arguments:
-            network: String, The network UUID
+            net: Reference to OpenStackNetwork object
 
         Returns:
             bool - True on success else False
 
         Note: This method returns True only if the given network has more than 3
         """
         try:
-            resp = self.driver_v2.network_connection.request(
```

Comment:
Heh, I did not realize we were already making some direct HTTP requests now, beyond what libcloud supports.